### PR TITLE
9 output object

### LINKED ISSUE #9: output object
create an output object with a state attribute (to capture the model's states as it runs) and methods for displaying data.

### LINKED ISSUE #12: remove observation() method from Model
Observation() is not needed since we are now storing the states in an Output class, and we can observe whatever we want in the states attribute of the Output class.

### PR DESCRIPTION
**Pull request by atiumcache**
closes #12 
closes #9 

See docs/output_class-2024-06-16 for an example of the Output class implementation.

Currently, the Output object stores the state and defines a plotting method for displaying a basic plot. More functionality can be added when we have a better idea of what we want (export to csv, different types of plots, etc.)

### CODE REVIEW

**Comment by Averydx** [comment]
A couple things, I would indicate return types for all the functions in abstract.py, even the constructor just for consistency. I also think we should follow the convention of upper level directory name, "output", then followed by "abstract" and "implementations" with each class in its own file for readability. That way we don't have one expanding file. Just some ideas.

**Comment by Averydx** [comment]
Oh also, I do think the observation function or at least a field in the output object is necessary. Not always the case that observation information can be extracted directly from the state variables.

File: sphere/model/abstract/model.py [hunk 1]
```diff
@@ -25,10 +25,6 @@ def __init__(self, params: Parameters, solver: ODESolver):
     def state_transition(self, state: ArrayLike, t: int) -> Array:
         pass
 
-    @abstractmethod
-    def observation(self, state: ArrayLike, t: int) -> Array:
```
Comment (Averydx) [inline sphere/model/abstract/model.py]: I think we might want to consider keeping this.

**Comment by atiumcache** [comment]
I agree with your comments, I'll make these changes.

File: sphere/model/abstract/model.py (hunk 1 above)
Comment (atiumcache) [reply sphere/model/abstract/model.py]: i added it back in new commit.

**Comment by atiumcache** [comment]
updated with a new commit that addresses/fixes your comments @Averydx.

**Review by Averydx — APPROVED** [review]
Looks great!

**Pull request merged**